Tianrui luo branch new

Tianrui-Luo opened the pull request.

JohnLYYN123 commented:
Please double check the conflict issue and use the most-updated API endpoint version from main branch

JohnLYYN123 reviewed (approved):
Tianrui has added Logging system into APIs and they are controlled by admin.py. The logging system tracks the status and the execution of DB related operations

Tianrui-Luo commented:
In this pull request, I added the logger function (admin.py) to record error messages in logs/operation.log. Also added partial test case for general information testing.
